primitives/keystore/src/lib.rs

```diff
@@ -274,8 +275,9 @@ pub trait SyncCryptoStore: CryptoStore + Send + Sync {
 
 	/// Checks if the private keys for the given public key and key type combinations exist.
 	///
-	/// Returns `true` iff all private keys could be found.
-	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool;
+	/// Returns the indices of the matching keys in `public_keys`. Call `is_empty()` on the returned
+	/// list to check if no keys match.
+	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> Vec<usize>;
```

Comment:
First, sorry for the late reply.

Now after enough time, I actually think we should return some enum here.

```
enum HasKeys {
    None,
    Found(Vec<usize>),
}

impl HasKeys {
    fn found_any(&self) -> bool {
        matches!(self, Self::Found(_))
    }

    fn into_found(self) -> impl Iterator<Item = usize> {
         match self {
             Self::None => Vec::new().into_iter(),
             Self::Found(found) => found.into_iter(),
        }
    }
    }
}
```

Probably the api needs to be improved, but you should get the rough idea.

Reply:
Looks sleek, maybe we can also add a `found_all` function, if you think that's useful?

Reply:
Yeah.